### PR TITLE
re #894: Fix bad frame rate handling in Azure Kinect

### LINKED ISSUE #894: Incorrect acquisition rate in Azure Kinect device
Currently, the acquisition and update rates are initialized from the c++ enum value `k4a_fps_t` (ie: 0,1 or 2 fps) instead of the corresponding frame rate (5, 15 or 30 fps).

Thanks to @IliasMAOUDJ for reporting this major issue.

### PR DESCRIPTION
**Pull request by kenavolic**
* Fix issue
* Add warning for frame rate ambiguity

Fixes #894

### CODE REVIEW

File: src/PlusDataCollection/AzureKinect/vtkPlusAzureKinect.cxx [hunk 1]
```diff
@@ -197,8 +200,15 @@ class vtkPlusAzureKinect::vtkInternal
       depthStream->Height = rgbStream->Height;
     }
 
-    External->InternalUpdateRate = DeviceConfig.camera_fps;
-    External->AcquisitionRate = DeviceConfig.camera_fps;
+    std::sort(std::begin(frameRates), std::end(frameRates));
+    frameRates.erase(std::unique(std::begin(frameRates), std::end(frameRates)), std::end(frameRates));
+    
+    if (frameRates.size() > 1) {
```
Comment (Sunderlandkyl) [inline src/PlusDataCollection/AzureKinect/vtkPlusAzureKinect.cxx:206]: Bracket on new line.

File: src/PlusDataCollection/AzureKinect/vtkPlusAzureKinect.cxx [hunk 2]
```diff
@@ -146,7 +146,8 @@ class vtkPlusAzureKinect::vtkInternal
   PlusStatus UpdateConfig()
   {
     bool status{false};
-    for (auto& Stream : StreamList)
+    std::vector<uint32_t> frameRates;
+    for (auto &Stream : StreamList)
```
Comment (Sunderlandkyl) [inline src/PlusDataCollection/AzureKinect/vtkPlusAzureKinect.cxx:150]: Stream should be lower-case.

File: src/PlusDataCollection/AzureKinect/vtkPlusAzureKinect.cxx [hunk 3]
```diff
@@ -197,8 +200,15 @@ class vtkPlusAzureKinect::vtkInternal
...
-    External->InternalUpdateRate = DeviceConfig.camera_fps;
-    External->AcquisitionRate = DeviceConfig.camera_fps;
+    std::sort(std::begin(frameRates), std::end(frameRates));
+    frameRates.erase(std::unique(std::begin(frameRates), std::end(frameRates)), std::end(frameRates));
+    
+    if (frameRates.size() > 1) {
+      LOG_WARNING("Detecting different frame rates for the streams: selecting the lowest rate (" << frameRates[0] << " fps)");
+    }
+
+    External->InternalUpdateRate = frameRates[0];
```
[the remark this answers is not in the document]
Comment (kenavolic) [reply src/PlusDataCollection/AzureKinect/vtkPlusAzureKinect.cxx]: Fixed

File: src/PlusDataCollection/AzureKinect/vtkPlusAzureKinect.cxx (hunk 2 above)
Comment (kenavolic) [reply src/PlusDataCollection/AzureKinect/vtkPlusAzureKinect.cxx]: Fixed

**Comment by Sunderlandkyl** [comment]
Looks good, thanks!

**Pull request merged**